flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java

```diff
@@ -161,6 +161,8 @@ public CliFrontend(String configDir) throws Exception {
 				"filesystem scheme from configuration.", e);
 		}
 
+		this.config.setString(ConfigConstants.FLINK_BASE_DIR_PATH_KEY, configDirectory.getAbsolutePath());
```

Comment:
Good catch, though the implementation takes care of handling base directory and the immediate "conf" directory. Will change it to point to parent directory.